frontend/src/components/orders/OrderDocuments.vue

```diff
@@ -4,48 +4,70 @@ import Dialog from "primevue/dialog";
 import ProgressSpinner from "primevue/progressspinner";
 import { ref } from "vue";
 
-import { getDocumentsByOrder } from "@/backendClient";
+import { getDocumentsByOrder, getDocumentsBySubOrder } from "@/backendClient";
+import { DocumentKind } from "@/global/types/appTypes";
 import {
   type InvoiceDocument,
   type OfferDocument,
   type OverdueNoticeDocument,
 } from "@/global/types/entities";
 
 const props = defineProps<{
-  id: string;
+  id: string | number;
+  kind?: DocumentKind;
 }>();
 
 let visible = ref(false);
 let documents = ref<(OfferDocument | OverdueNoticeDocument | InvoiceDocument)[]>([]);
 let isLoading = ref<boolean>(false);
 
-async function getDocuments() {
+async function getAllDocuments() {
   isLoading.value = true;
   documents.value = await getDocumentsByOrder(props.id as string);
   isLoading.value = false;
 }
-async function openDocumentsList() {
+async function openAllDocumentsList() {
   visible.value = true;
-  getDocuments();
+  getAllDocuments();
 }
+async function getItemDocuments() {
+  isLoading.value = true;
+  documents.value = await getDocumentsBySubOrder(props.id as number, props.kind as DocumentKind);
+  isLoading.value = false;
+}
+async function openItemDocumentsList() {
+  visible.value = true;
+  getItemDocuments();
+}
+console.log(DocumentKind);
 </script>
 
 <template>
-  <div class="font-bold my-2">Dokumente</div>
-  <div>
+  <div v-if="!kind" class="font-bold my-2">Dokumente</div>
+  <div class="flex flex-row gap-4">
     <Button
-      @click="openDocumentsList"
+      v-if="!kind"
+      @click="openAllDocumentsList"
       label="Alle Auftragsdokumente anzeigen"
       severity="secondary"
       outlined
       size="small"
     />
+    <Button
+      v-else
+      @click="openItemDocumentsList"
+      label="Show all SubItem Documents"
+      severity="secondary"
+      outlined
+      size="small"
+    />
```

Comment:
This component seems a bit weird and contains a bit much branching logic.

It seems the "OrderDocuments" component mainly contains two buttons (of which only ever one is shown) and then documents are selected.
Maybe it makes more sense to extract the buttons out of this component and only have the "document selection modal" as a component?